src/api_manager/service_control/proto.cc

```diff
@@ -1224,40 +1179,18 @@ Status Proto::FillReportRequest(const ReportRequestInfo& info,
         }
       }
 
-      // Set all labels.
-      for (auto it = by_consumer_labels_.begin(),
-                end = by_consumer_labels_.end();
-           it != end; it++) {
-        const SupportedLabel* l = *it;
-        if (l->set) {
-          status = (l->set)(*l, info, labels);
-          if (!status.ok()) return status;
-        }
-      }
-
-      // Not to send consumer metrics if api_key is empty.
-      // api_key is empty in one of following cases:
-      // 1) api_key is not provided,
-      // 2) api_key is invalid determined by the server from the Check call.
-      // 3) the service is not activated for the consumer project.
-      bool send_consumer_metric = !info.api_key.empty();
-
       // Populate all metrics.
```

Comment:
what about the labels?  maybe we can copy all labels from old operation and add extra one on top of it.